### ISSUE TITLE
select a group of vessels to track

### ISSUE THREAD

**Issue by g7ruh** [jvde-github/AIS-catcher#253]
Jasper,

As you know, I live in an area renowned for yacht racing.

It is interesting to watch the tactics develop. 

Would it be possible to enable a group of vessels to have tracking enabled? E.G. use a mouse button to create a rectangle and set tracking on for all vessels within that rectangle.

select an area as below

![select rectangle-all boats within aiscatcher-edit01](https://github.com/jvde-github/AIS-catcher/assets/5949609/fe7f261b-f894-4a70-9de6-7cb0f5051ca5)

and enable tracking for all vessels:

![race-tracks-selected-individually](https://github.com/jvde-github/AIS-catcher/assets/5949609/a35d761a-d1f9-4406-8fca-17f76f47a22a)

Of course the idea came when I had to select all one at a time!

The track length needs to be variable, it is too short for this use case:

![group-select-track-length-needs increase](https://github.com/jvde-github/AIS-catcher/assets/5949609/bdae3660-b65a-4c62-a290-1f3e37804571)

and from OpenCPN which allows track length to be configured

![longer track-length-opencpn-example-02](https://github.com/jvde-github/AIS-catcher/assets/5949609/b1335924-0b16-4fae-80c0-fad6c30280b1)

Thanks for considering this,
regards
Roger

**Comment by g7ruh** [comment]
Of course, the arrow indicating speed and direction on the OpenCPN map adds more to the experience, as a real-time status

Roger

**Comment by g7ruh** [comment]
![image](https://github.com/jvde-github/AIS-catcher/assets/5949609/3d162ad1-7b91-4f9f-a77f-81c7aeb8ace4)

the highlight when hovering over the vessel icon is very good at showing the individual tactics.

Roger

**Comment by g7ruh** [comment]
Jasper,

the race in the example went on for most of the afternoon, so the tracks were very long. if a bit thick on the default OpenCPN track width, it made a 'mess' of confusion on the screen, however your feature to highlight when hovering on a vessel picks out the detail nicely.

To that end, I think that the length of track (time-wise) needs to be a configurable setting, for normal use as it is or a bit longer is OK, for tracking a race and tactics, it needs to be a lot longer, but configurable.

Maybe AIS-catcher needs to have the option to record tracks for the "selected" rectangle or all tracks for e.g. 24 hours so the tracks are recoverable, post event, at least for a short while. E.G. playing back this afternoon's Cowes Week races: there are many separate races happening in The Solent at the same time. That option may win you some users in the yacht racing world too. Replay analysis tool, the tactician's friend!!!!!!!

Roger

**Comment by kit-powell** [comment]
This is right up my alley, as I want to do something similar (not for racing per se, but for fishing)

I’m not sure how fast I can get it done but if y’all would like I’d work on a companion platform to create a sort of geo-temporal fence and replay the tracks (probably leaflet based) with varying speed and chart/map/imagery)

Though I freely admit my github experience is limited to one project I put up years ago

**Comment by gharbarka** [comment]
hi,

as a yacht racing enthusiast I am interested in an application using ais-catcher for tracking and recording a list of boats based on names or MMSI number for playback later offline for the purpose of analysing strategies of individual boats. This or something similar is intimated in the above discussion. Has any progress been achieved since the last message was posted a year ago?  Or perhaps a related project may have been created. Thanks for any insight.

**Comment by gharbarka** [comment]
Thank you very much for your speedy reply. First of all allow me to congratulate you for a very nifty piece of software.

Yes I have read issue: #341 but it seems to add little to the problem. I have also read the wiki section related to the use of PostgreSQL which is interesting but to me somewhat of an overkill for the purpose i have in mind. Also, as I understand it you still have to process that database further to get to an end result. The issue of needing to be connected to the internet to use the database would also be an issue for on the water analysis.

I wonder if it is possible to send the  !AIVDM output to a decoder which can extract the positions and perhaps the speed of selected boats  with a time stamp to a simple .csv file for further processing.

**Comment by gharbarka** [comment]
Indeed I had already started going down that route. I am currently learning how to process json files for futher processing in Python. The json concept is new to me.

What would be more useful would be the possibility of processing the json format in real time and then only save fields of interested related to a preposted list of MMSI numbers.

**Comment by jvde-github** [comment]
Roger, to track vessels in an area has been added.

<img width="334" height="154" alt="Image" src="https://github.com/user-attachments/assets/0a0c6128-bc00-4adc-b19d-904bada75eb7" />

**Comment by g7ruh** [comment]
Jasper,

Many thanks, I have given it a brief test, looks good. I will test tomorrow. Weekends in The Solent are very busy and race starts usually have tight clusters of boats as they gather at the race start line. idea for further testing.

Roger

**Comment by g7ruh** [comment]
Jasper,

A few thoughts based on the very busy day on The Solent, so far:

For context, there are currently 362 Class B vessels moving in this area (quite a lot) but on some big race days there are more than 600! About 60 Class A (many local, small utility / tug type). Probably at least half are located some way south in the English Channel so will have much less impact signal strength-wise (see below). There is currently some atmospheric enhancement, hence a higher number than usual.

It would be helpful to be able to 'delete' all the previous track history when creating a local group. Otherwise the travel from home base to race start adds 'noise' to the area of interest. This area is the grouping just before race start until end of race.

<img width="231" height="209" alt="Image" src="https://github.com/user-attachments/assets/3f33a9c3-327b-4615-843a-078a9ff4323e" />

This is a somewhat larger group than I would typically think of tracking, but still a valid scenario. You can see the travel tracks coming to the middle of the grouping from the NW and one from SW. It is hard to make out the actual race track as a lot is covered by the pre-race history.

I appreciate that a zoom in will help (and does) make track more obvious, but I use this small screen grab to illustrate the 'issue'.

There may well be lots of gaps in the tracks, partly caused by my receiver location but also a lot are caused by the Class A vessels "drowning out" the Class B signals with the much greater RF power used in their transmissions. Currently two large vessels with very high antenna, for example, as well as local ferry traffic, same RF power but much lower antenna. We cannot do anything about that! I mention it for context in this area.

thanks for considering,
Roger

**Comment by jvde-github** [comment]
<img width="433" height="196" alt="Image" src="https://github.com/user-attachments/assets/223c87cc-7326-4128-a978-95848b1a607c" />

Maybe this option would help?

**Comment by g7ruh** [comment]
I have tested some more. The feature works OK, but 'Reset Tracks (from now)' resets all tracks. There may be some occasions when you want the historical tracks for some vessels, but want to reset only the ones e.g. involved in a race so that you can remove the noise: the journey to the race start line. If an area is set, can just all the vessels included in that area have the tracks reset, not all history?

I note you can set multiple areas (but reset all tracks) so that might complicate things unless you can reset by area.

I got diverted at the weekend by family matters so could not test the race start scenario very much. I shall do that next weekend. But: it looks like it does what I requested, thanks Jasper.
